crates/ruff/src/importer/mod.rs

```diff
@@ -234,6 +260,48 @@ impl<'a> Importer<'a> {
     }
 }
 
+#[derive(Debug)]
+enum ImportStyle {
+    /// Import the symbol using the `import` statement (e.g. `import foo; foo.bar`).
+    Import,
+    /// Import the symbol using the `from` statement (e.g. `from foo import bar; bar`).
+    ImportFrom,
+}
+
+/// A symbol that can be imported from a module.
+#[derive(Debug)]
+pub(crate) struct ImportableSymbol<'a> {
```

Comment:
I like the `API` but find the name `ImportableSymbol` somewhat confusing because two symbols for the same `module` and `member` should compare equal, even if they use different import styles. 

I guess another way to think about it. The qualifying information that defines a symbol is the `module` and the `member`. The style only denotes on how a missing import should be added.

Should we just call this `MemberImport` (or `ImportMember`) or even `Import`?

Reply:
`ImportRequest` maybe? Since it's less describing a symbol and more "please make an import for this symbol with this preference"